### PR TITLE
[ADD] pylint-odoo: Adding new check unnecessary-attribute-noupdate

### PR DESCRIPTION
**Pull request by JesusZapata**
This check detected the unnecessary noupdate attribute `<odoo noupdate=0>/<data noupdate=0>` because the default value of attribute no update is zero

Example of the detected code 

```xml
<?xml version="1.0" encoding="utf-8"?>                                          
<odoo noupdate="0">
    ...
</odoo>    
```
```xml
<?xml version="1.0" encoding="utf-8"?>                                          
<odoo>
    <data noupdate="0">
    ...
    </data>
</odoo>    
```

### CODE REVIEW

**Comment by moylop260** [comment]
@JesusZapata 
A valid case is:

```xml
<?xml version="1.0" encoding="utf-8"?>                                          
<odoo>
    <data noupdate="0">
    ...
    </data>
    <data noupdate="1">
    ...
    </data>
</odoo>
```

Based on comment: https://github.com/odoo/odoo/pull/18144#discussion_r127104759
(Just if is the same file)

**Comment by JesusZapata** [comment]
@moylop260 
I add one file with two tag `noupdate` into the same file

Similar to 

```xml
<?xml version="1.0" encoding="utf-8"?>
<odoo>
    <data noupdate="0">
    </data>
    ...
    <data noupdate="1">
    </data>
</odoo>
```

**Comment by moylop260** [comment]
Consider the case:
```xml
<?xml version="1.0" encoding="utf-8"?>
<odoo>
    <data noupdate="0">
    </data>
    ...
    <record...
</odoo>
```

**Comment by JesusZapata** [comment]
@moylop260 
I supported your new case.

File: pylint_odoo/test_repo/broken_module2/tests/data/odoo_data_noupdate_0_2.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="0">
+    </data>
+    <record>
```
Comment (JesusZapata) [inline pylint_odoo/test_repo/broken_module2/tests/data/odoo_data_noupdate_0_2.xml]: This file not should be raise the `deprecated-data-xml-node` check.
I go to work in that fix in another PR

**Comment by JesusZapata** [comment]
@moylop260 
This PR is RED for another check `deprecated-data-xml-node`
I go to work in that fix in another PR

**Comment by JesusZapata** [comment]
@moylop260 
Now this pull request is green

File: pylint_odoo/test_repo/broken_module2/tests/data/odoo_data_noupdate_0_2.xml (hunk 1 above)
Comment (moylop260) [reply pylint_odoo/test_repo/broken_module2/tests/data/odoo_data_noupdate_0_2.xml]: It did you work for oca
is not?

Comment (JesusZapata) [reply pylint_odoo/test_repo/broken_module2/tests/data/odoo_data_noupdate_0_2.xml]: @moylop260 
Yes, this is the another pull request https://github.com/OCA/pylint-odoo/pull/156 
In this pull request the false positive on `deprecated-data-xml-node` was fixed 

Sorry for not report this change

**Comment by moylop260** [comment]
Closing since that there are not new comments since a while

**Pull request closed without merging**